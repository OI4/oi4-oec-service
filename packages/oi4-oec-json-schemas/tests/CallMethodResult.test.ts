import { matchersWithOptions } from 'jest-json-schema'
import schema from '../src/schemas/CallMethodResult.schema.json'
import BaseDataType from '../src/schemas/BaseDataType.schema.json'
import uint32 from '../src/schemas/dataTypes/uint32.schema.json'

expect.extend(
  matchersWithOptions({
    // Loading in a schema which is comprised only of definitions,
    // which means specific test schemas need to be created.
    // This is good for testing specific conditions for definition schemas.
    schemas: [BaseDataType, uint32],
    verbose: true,
  })
)

describe('CallMethodResult schema', () => {
  it('validate schema', () => {
    expect(schema).toBeValidSchema()
  })

  it('test valid values', () => {
    expect({
      StatusCode: 0,
      InputArgumentResults: [],
      OutputArguments: [],
    }).toMatchSchema(schema)
    expect({
      StatusCode: 4294967295,
      InputArgumentResults: [],
      OutputArguments: [],
    }).toMatchSchema(schema)
  })

  it('test invalid values', () => {
    const invalidValues = [
      {},
      { InputArgumentResults: [], OutputArguments: [] },
      { StatusCode: 'a', InputArgumentResults: [], OutputArguments: [] },
      { StatusCode: -1, InputArgumentResults: [], OutputArguments: [] },
      { StatusCode: 4294967296, InputArgumentResults: [], OutputArguments: [] },
    ]

    invalidValues.forEach((value, index) => {
      expect(value).not.toMatchSchema(schema)
      expect(() => {
        expect(value).toMatchSchema(schema)
      }).toThrowErrorMatchingSnapshot(index.toString())
    })
  })
})
