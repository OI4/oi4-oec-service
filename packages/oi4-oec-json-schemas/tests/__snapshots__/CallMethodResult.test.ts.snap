// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CallMethodResult schema test invalid values: 0 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toMatchSchema([22m[32mschema[39m[2m)[22m

received
[31m  must have required property 'StatusCode'[39m
[31m    Path:     CallMethodResult.schema.json#/required[39m
[31m[39m[31m  must have required property 'InputArgumentResults'[39m
[31m    Path:     CallMethodResult.schema.json#/required[39m
[31m[39m[31m  must have required property 'OutputArguments'[39m
[31m    Path:     CallMethodResult.schema.json#/required[39m
[31m[39m"
`;

exports[`CallMethodResult schema test invalid values: 1 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toMatchSchema([22m[32mschema[39m[2m)[22m

received
[31m  must have required property 'StatusCode'[39m
[31m    Path:     CallMethodResult.schema.json#/required[39m
[31m[39m"
`;

exports[`CallMethodResult schema test invalid values: 2 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toMatchSchema([22m[32mschema[39m[2m)[22m

received
[31m  /StatusCode must be integer[39m
[31m    Received: [33m<string>[39m[31m a[39m
[31m    Path:     CallMethodResult.schema.jsondataTypes/uint32.schema.json/type[39m
[31m[39m"
`;

exports[`CallMethodResult schema test invalid values: 3 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toMatchSchema([22m[32mschema[39m[2m)[22m

received
[31m  /StatusCode must be >= 0[39m
[31m    Received: [33m<number>[39m[31m -1[39m
[31m    Path:     CallMethodResult.schema.jsondataTypes/uint32.schema.json/minimum[39m
[31m[39m"
`;

exports[`CallMethodResult schema test invalid values: 4 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toMatchSchema([22m[32mschema[39m[2m)[22m

received
[31m  /StatusCode must be <= 4294967295[39m
[31m    Received: [33m<number>[39m[31m 4294967296[39m
[31m    Path:     CallMethodResult.schema.jsondataTypes/uint32.schema.json/maximum[39m
[31m[39m"
`;
